### PR TITLE
Leaderboards on the hub: Vectara

### PR DESCRIPTION
**Pull request by clefourrier**
Also added a specific tag for the openllmleaderboard blogs in the general metadata.

### CODE REVIEW

File: leaderboards-on-the-hub-vectara.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+# Setting up HHEM with the LLM leaderboard template
+To set up the Vectara HHEM leaderboard, we had to follow a few steps, adjusting the HF leaderboard template code to our needs:
+1. After cloning the space repository to our own organization, we created two associated datasets, “requests” and “results”; these datasets maintain the requests submitted by users for new LLMs to evaluate, and the results of such evaluation, respectively.
+2. We populated the results dataset with existing results from the initial launch, and updated the “About” and “Citations” sections.
+
+For a simple leaderboard, where evaluations results are pushed by your backend to the results dataset, that’s all you need!
+
+As our evaluation is more complex, we then customized the source code to fit the needs of the HHEM leaderboard - here are the details:
+1. `leaderboard/src/backend/model_operations.py`: This file contains two primary classes - SummaryGenerator and EvaluationModel.
```
Comment (ofermend) [inline leaderboards-on-the-hub-vectara.md:42]: Perhaps highlight instances of SummaryGenerator and EvaluationModel with backtick to note they are code?
Also the (a) and (b) subsections just show as text after the (1) bullet and do not show as sub-bullets.

File: _blog.yml [hunk 2]
```diff
@@ -2431,6 +2431,7 @@
     - research
     - nlp
     - evaluation
+    - openllmleaderboard
```
Comment (julien-c) [inline _blog.yml:2434]:
```suggestion
    - open-llm-leaderboard
```

maybe more readable like this (unless you already used it before?)

**Review by ofermend — APPROVED** [review]
Left a few minor comments to fix for good display, otherwise LGTM.

File: _blog.yml (hunk 2 above)
Comment (clefourrier) [reply _blog.yml:2434]: I'm OK with either, it's the first introduction of this tag :)

Comment (clefourrier) [reply _blog.yml]: Changed it :)

File: leaderboards-on-the-hub-vectara.md (hunk 1 above)
Comment (clefourrier) [reply leaderboards-on-the-hub-vectara.md]: Should be good now

File: _blog.yml [hunk 3]
```diff
@@ -3245,3 +3247,15 @@
     - optimization
     - llm
     - qlora
+
+- local: leaderboards-on-the-hub-vectara
```
Comment (pcuenca) [inline _blog.yml:3251]: Could we maybe try to shorten the URL?

File: leaderboards-on-the-hub-vectara.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+This has been of tremendous value to the open-source community, as it provides a way for practitioners to keep track of the best open-source models.
+
+In late 2023, at Vectara we introduced the [Hughes Hallucination Evaluation Model](https://huggingface.co/vectara/hallucination_evaluation_model) (HHEM), an open-source model for measuring the extent to which an LLM [hallucinates](https://vectara.com/glossary-of-llm-terms/#h-hallucinations-aka-llm-hallucinations) (generates text that is nonsensical or unfaithful to the provided source content). Covering both open source models like Llama2 or Mistral-7B as well as commercial models like OpenAI’s GPT-4, Anthropic Claude, or Google’s Gemini, this model highlighted the stark differences that currently exist between models in terms of their likelihood to hallucinate.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:17]:
```suggestion
In late 2023, at Vectara we introduced the [Hughes Hallucination Evaluation Model](https://huggingface.co/vectara/hallucination_evaluation_model) (HHEM), an open-source model for measuring the extent to which an LLM [hallucinates](https://vectara.com/glossary-of-llm-terms/#h-hallucinations-aka-llm-hallucinations) (generates text that is nonsensical or unfaithful to the provided source content). Covering both open source models like Llama 2 or Mistral 7B, as well as commercial models like OpenAI’s GPT-4, Anthropic Claude, or Google’s Gemini, this model highlighted the stark differences that currently exist between models in terms of their likelihood to hallucinate.
```

Maybe add links to the Hub for Llama 2 and Mistral, or to the project pages for the others.

File: leaderboards-on-the-hub-vectara.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+As we continue to add new models to HHEM, we were looking for an open-source solution to manage and update the HHEM leaderboard. 
+
+Quite recently, the Hugging Face leaderboard team released leaderboard templates ([here](https://huggingface.co/demo-leaderboard) and [here](https://huggingface.co/demo-leaderboard-backend)). These are lightweight versions of the Open LLM Leaderboard itself, which are both open-source and simpler to use than the original code. 
+
+Today we’re happy to announce the release of the [new HHEM leaderboard](https://huggingface.co/spaces/vectara/leaderboard), powered by the [HF leaderboard template](https://huggingface.co/demo-leaderboard-backend).
+
+# Vectara’s Hughes Hallucination Evaluation Model (HHEM)
+The Hughes Hallucination Evaluation Model (HHEM) Leaderboard is dedicated to assessing the frequency of hallucinations in document summaries generated by Large Language Models (LLMs) such as GPT-4, Google Gemini or Meta’s Llama2. To use it you can follow the instructions [here](https://huggingface.co/vectara/hallucination_evaluation_model).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:26]:
```suggestion
The Hughes Hallucination Evaluation Model (HHEM) Leaderboard is dedicated to assessing the frequency of hallucinations in document summaries generated by Large Language Models (LLMs) such as GPT-4, Google Gemini or Meta’s Llama 2. To use it you can follow the instructions [here](https://huggingface.co/vectara/hallucination_evaluation_model).
```

File: leaderboards-on-the-hub-vectara.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+By doing an open-source release of this model, we (at [Vectara](https://vectara.com)) aim to democratize the evaluation of LLM hallucinations, driving awareness to the differences that exist in LLM performance in terms of propensity to hallucinate.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:28]:
```suggestion
By doing an open-source release of this model, we at [Vectara](https://vectara.com) aim to democratize the evaluation of LLM hallucinations, driving awareness to the differences that exist in LLM performance in terms of propensity to hallucinate. 
```

File: leaderboards-on-the-hub-vectara.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+thumbnail: /blog/assets/leaderboards-on-the-hub/thumbnail.png
+authors:
+- user: ofermend
+  guest: true
+- user: minseokbae
+  guest: true
+- user: clefourrier
+---
+
+# Introduction
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:12]:
```suggestion
# A guide to setting up your own Hugging Face leaderboard: an end-to-end example with Vectara's hallucination leaderboard
```

File: leaderboards-on-the-hub-vectara.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+As we continue to add new models to HHEM, we were looking for an open-source solution to manage and update the HHEM leaderboard. 
+
+Quite recently, the Hugging Face leaderboard team released leaderboard templates ([here](https://huggingface.co/demo-leaderboard) and [here](https://huggingface.co/demo-leaderboard-backend)). These are lightweight versions of the Open LLM Leaderboard itself, which are both open-source and simpler to use than the original code. 
+
+Today we’re happy to announce the release of the [new HHEM leaderboard](https://huggingface.co/spaces/vectara/leaderboard), powered by the [HF leaderboard template](https://huggingface.co/demo-leaderboard-backend).
+
+# Vectara’s Hughes Hallucination Evaluation Model (HHEM)
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:25]:
```suggestion
## Vectara’s Hughes Hallucination Evaluation Model (HHEM)
```

File: leaderboards-on-the-hub-vectara.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+The Hughes Hallucination Evaluation Model (HHEM) Leaderboard is dedicated to assessing the frequency of hallucinations in document summaries generated by Large Language Models (LLMs) such as GPT-4, Google Gemini or Meta’s Llama2. To use it you can follow the instructions [here](https://huggingface.co/vectara/hallucination_evaluation_model).
+
+By doing an open-source release of this model, we (at [Vectara](https://vectara.com)) aim to democratize the evaluation of LLM hallucinations, driving awareness to the differences that exist in LLM performance in terms of propensity to hallucinate. 
+
+Our initial release of HHEM was a [Huggingface model](https://huggingface.co/vectara/hallucination_evaluation_model) alongside the [Github repository](https://github.com/vectara/hallucination-leaderboard) (which is still maintained), but we quickly realized that we needed a mechanism to allow new types of models to be evaluated. Using the HF leaderboard code template, we were able to quickly put together the new leaderboard that allows for dynamic updates, and we encourage the LLM community to submit new relevant models for HHEM evaluation.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:30]:
```suggestion
Our initial release of HHEM was a [Huggingface model](https://huggingface.co/vectara/hallucination_evaluation_model) alongside a [Github repository](https://github.com/vectara/hallucination-leaderboard), but we quickly realized that we needed a mechanism to allow new types of models to be evaluated. Using the HF leaderboard code template, we were able to quickly put together a new leaderboard that allows for dynamic updates, and we encourage the LLM community to submit new relevant models for HHEM evaluation.
```

File: leaderboards-on-the-hub-vectara.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+Side note: *On a meaningful side note to us here at Vectara, the HHEM was named after our peer Simon Hughes, who passed away in Nov. of 2023 without notice of natural causes; we decided to name it in his honor due to his [lasting legacy](https://diginomica.com/final-thoughts-memory-simon-mark-hughes-hallucination-research-pioneer) in this space.*
+
+# Setting up HHEM with the LLM leaderboard template
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:34]:
```suggestion
## Setting up HHEM with the LLM leaderboard template
```

File: leaderboards-on-the-hub-vectara.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+As we continue to add new models to HHEM, we were looking for an open-source solution to manage and update the HHEM leaderboard. 
+
+Quite recently, the Hugging Face leaderboard team released leaderboard templates ([here](https://huggingface.co/demo-leaderboard) and [here](https://huggingface.co/demo-leaderboard-backend)). These are lightweight versions of the Open LLM Leaderboard itself, which are both open-source and simpler to use than the original code. 
+
+Today we’re happy to announce the release of the [new HHEM leaderboard](https://huggingface.co/spaces/vectara/leaderboard), powered by the [HF leaderboard template](https://huggingface.co/demo-leaderboard-backend).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:23]: Still not public, is it?

File: leaderboards-on-the-hub-vectara.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+Side note: *On a meaningful side note to us here at Vectara, the HHEM was named after our peer Simon Hughes, who passed away in Nov. of 2023 without notice of natural causes; we decided to name it in his honor due to his [lasting legacy](https://diginomica.com/final-thoughts-memory-simon-mark-hughes-hallucination-research-pioneer) in this space.*
+
+# Setting up HHEM with the LLM leaderboard template
+To set up the Vectara HHEM leaderboard, we had to follow a few steps, adjusting the HF leaderboard template code to our needs:
+1. After cloning the space repository to our own organization, we created two associated datasets, “requests” and “results”; these datasets maintain the requests submitted by users for new LLMs to evaluate, and the results of such evaluation, respectively.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:36]:
```suggestion
1. After cloning the space repository to our own organization, we created two associated datasets: “requests” and “results”; these datasets maintain the requests submitted by users for new LLMs to evaluate, and the results of such evaluations, respectively.
```

I'd maybe link to these datasets (when they are up) so users interested can explore what they look like.

File: leaderboards-on-the-hub-vectara.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    a. The `SummaryGenerator` generates summaries based on your evaluation dataset and calculates metrics like Answer Rate and Average Summary Length.
+    b. The `EvaluationModel` loads our proprietary Hughes Hallucination Evaluation Model (HHEM) to assess these summaries, yielding metrics such as Factual Consistency Rate and Hallucination Rate.
+2. `leaderboard/src/backend/evaluate_model.py`: defines the `Evaluator` class which utilizes both `SummaryGenerator` and `EvaluationModel` to compute and return results in JSON format. 
+3. `leaderboard/src/backend/run_eval_suite.py`: contains a function run_evaluation that leverages `Evaluator` to obtain and upload evaluation results to Vectara (in the “results” dataset mentioned above).
+4. `leaderboard/main_backend.py`: Manages pending evaluation requests and executes auto evaluations using aforementioned classes and functions. It also includes an option for users to replicate our evaluation results.
+
+The final source code is available in our [HHEM leaderboard repository](https://huggingface.co/spaces/vectara/leaderboard).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:49]:
```suggestion
The final source code is available in [the Files tab](https://huggingface.co/spaces/vectara/leaderboard/tree/main) of our [HHEM leaderboard repository](https://huggingface.co/spaces/vectara/leaderboard).
```

(and/or link the individual files above)

File: leaderboards-on-the-hub-vectara.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+With all these changes, we now have the evaluation pipeline ready to go, and easily deployable as a Huggingface space.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:50]:
```suggestion
With all these changes, we now have the evaluation pipeline ready to go, and easily deployable as a Huggingface Space.
```

File: leaderboards-on-the-hub-vectara.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+
+# Summary
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:52]:
```suggestion
## Summary
```

File: leaderboards-on-the-hub-vectara.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+# Setting up HHEM with the LLM leaderboard template
+To set up the Vectara HHEM leaderboard, we had to follow a few steps, adjusting the HF leaderboard template code to our needs:
+1. After cloning the space repository to our own organization, we created two associated datasets, “requests” and “results”; these datasets maintain the requests submitted by users for new LLMs to evaluate, and the results of such evaluation, respectively.
+2. We populated the results dataset with existing results from the initial launch, and updated the “About” and “Citations” sections.
+
+For a simple leaderboard, where evaluations results are pushed by your backend to the results dataset, that’s all you need!
+
+As our evaluation is more complex, we then customized the source code to fit the needs of the HHEM leaderboard - here are the details:
+1. `leaderboard/src/backend/model_operations.py`: This file contains two primary classes - `SummaryGenerator` and `EvaluationModel`.
+    a. The `SummaryGenerator` generates summaries based on your evaluation dataset and calculates metrics like Answer Rate and Average Summary Length.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:43]: Is the evaluation dataset a HF dataset?

File: leaderboards-on-the-hub-vectara.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+    b. The `EvaluationModel` loads our proprietary Hughes Hallucination Evaluation Model (HHEM) to assess these summaries, yielding metrics such as Factual Consistency Rate and Hallucination Rate.
+2. `leaderboard/src/backend/evaluate_model.py`: defines the `Evaluator` class which utilizes both `SummaryGenerator` and `EvaluationModel` to compute and return results in JSON format. 
+3. `leaderboard/src/backend/run_eval_suite.py`: contains a function run_evaluation that leverages `Evaluator` to obtain and upload evaluation results to Vectara (in the “results” dataset mentioned above).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:46]:
```suggestion
3. `leaderboard/src/backend/run_eval_suite.py`: contains a function `run_evaluation` that leverages `Evaluator` to obtain and upload evaluation results to the `results` dataset mentioned above, causing them to appear in the leaderboard.
```

**Review by pcuenca — COMMENTED** [review]
Very cool!

File: _blog.yml (hunk 3 above)
Comment (clefourrier) [reply _blog.yml:3251]: I wanted to make it a series, `leaderboards-on-the-hub-leaderboardname` - what would you suggest for the beginning of the name?

File: leaderboards-on-the-hub-vectara.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+By doing an open-source release of this model, we (at [Vectara](https://vectara.com)) aim to democratize the evaluation of LLM hallucinations, driving awareness to the differences that exist in LLM performance in terms of propensity to hallucinate. 
+
+Our initial release of HHEM was a [Huggingface model](https://huggingface.co/vectara/hallucination_evaluation_model) alongside the [Github repository](https://github.com/vectara/hallucination-leaderboard) (which is still maintained), but we quickly realized that we needed a mechanism to allow new types of models to be evaluated. Using the HF leaderboard code template, we were able to quickly put together the new leaderboard that allows for dynamic updates, and we encourage the LLM community to submit new relevant models for HHEM evaluation.
+
+Side note: *On a meaningful side note to us here at Vectara, the HHEM was named after our peer Simon Hughes, who passed away in Nov. of 2023 without notice of natural causes; we decided to name it in his honor due to his [lasting legacy](https://diginomica.com/final-thoughts-memory-simon-mark-hughes-hallucination-research-pioneer) in this space.*
```
Comment (pcuenca) [inline leaderboards-on-the-hub-vectara.md:32]:
```suggestion
<br>
<div style="background-color: #e6f9e6; padding: 16px 32px; outline: 2px solid; border-radius: 10px;">
On a meaningful side note to us here at Vectara, the HHEM was named after our peer Simon Hughes, who passed away in Nov. of 2023 without notice of natural causes; we decided to name it in his honor due to his [lasting legacy](https://diginomica.com/final-thoughts-memory-simon-mark-hughes-hallucination-research-pioneer) in this space.
</div>
```

Just a suggestion if you want it to show inside a box like here: https://huggingface.co/blog/introducing-private-hub

Comment (clefourrier) [reply leaderboards-on-the-hub-vectara.md:32]: Looks nice

File: leaderboards-on-the-hub-vectara.md (hunk 16 above)
Comment (clefourrier) [reply leaderboards-on-the-hub-vectara.md:43]: cc @ofermend ?

File: leaderboards-on-the-hub-vectara.md (hunk 11 above)
Comment (clefourrier) [reply leaderboards-on-the-hub-vectara.md:23]: cc @ofermend - when do you want to pass it to public?

File: _blog.yml (hunk 3 above)
Comment (clefourrier) [reply _blog.yml:3251]: I could go for `lb-on-the-hub-` for ex (but I find it meh), or `hub-leaderboards-` ?

Comment (pcuenca) [reply _blog.yml:3251]: haha, `lb-on-the-hub-` is ugly I agree. I didn't know this was meant to be a series, `hub-leaderboards-` works but is not that much shorter, so let's use what you think is clearer.

Comment (clefourrier) [reply _blog.yml:3251]: let's keep leaderboards-on-the-hub then :)

Comment (julien-c) [reply _blog.yml:3251]: sounds good!

File: leaderboards-on-the-hub-vectara.md (hunk 11 above)
Comment (ofermend) [reply leaderboards-on-the-hub-vectara.md:23]: public now!

File: leaderboards-on-the-hub-vectara.md (hunk 16 above)
Comment (clefourrier) [reply leaderboards-on-the-hub-vectara.md:46]: It's a private proprietary dataset, I'll edit the blog

**Review by pcuenca — APPROVED** [review]

**Pull request merged**